### PR TITLE
isisdataeval Updates, fixes and additional documentation

### PR DESCRIPTION
**Pull request by KrisBecker**
Apologies I missed the merge of #5111 with a few, but perhaps helpful, updates.

isisdata_mockup.py - Added —tojson and —hasher parameters to improve utility

make_isisdata_mockup.sh - Fixed path to test data and added new parameters to the isisdata_mockup.py introduced in this PR

README.md - improved/clarified/updated documentation

## Related Issue

#5111 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by KrisBecker** [comment]
Question: Is it recommended/possible to install `isisdata_mockup.py` in a bin-type directory or someplace where its easier to access/use?

File: isis/src/system/apps/isisdataeval/tools/README.md [hunk 1]
```diff
@@ -17,29 +17,31 @@ The ISISDATA mockup system is a complete copy of every directory and every file
...
+  -h, --help            show this help message and exit
+  --isisdata ISISDATA   ISISDATA directory root
+  --outpath OUTPATH     NEW ISISDATA directory to create
+  --ghostdir GHOSTDIR   Replaces the --isisdata path with this string to ghost the actual input directory path
+  --hasher {md5,sha1,sha256,all}
+                        Add desired/supported hash algorithms to compute
+  --saveconfig, -s      Retain *.db, *.conf files rather than replace with processs info
+  --tojson TOJSON       Write all ISISDATA file information to output file in JSON format
+  --dryrun, -n          Only print --outpath actions but do not execute
+  --verbose, -v         Verbose output
```
Comment (Kelvinrr) [inline isis/src/system/apps/isisdataeval/tools/README.md]: could we just avoid putting in-the-weeds parameter details in the readme and instead instruct to run the command? this way we don't have to keep this in sync.

File: isis/src/system/apps/isisdataeval/tools/README.md [hunk 2]
```diff
@@ -63,53 +70,337 @@ Finally, by setting --ghostdir '$ISISDATA', this now provides a connection to
...
+And here is the contents of the output JSON file, `isisdata_base_dem_md5.json`:
+```
+{
+  "program": {
+    "name": "isisdata_mockup.py",
+    "version": "0.2",
+    "date": "2023-03-29",
+    "runtime": "2023-03-29T21:02:03.721127 UTC",
+    "endtime": "2023-03-29T21:02:39.887436 UTC",
+    "elapsedtime": "0:00:36.166309",
```
Comment (Kelvinrr) [inline isis/src/system/apps/isisdataeval/tools/README.md]: could we replace this with a truncated version so it isn't as verbose?

**Review by Kelvinrr — COMMENTED** [review]
some small comments. 

Partly also my bad for being a little hasty with wanting to merge that PR.

File: isis/src/system/apps/isisdataeval/tools/README.md (hunk 1 above)
Comment (KrisBecker) [reply isis/src/system/apps/isisdataeval/tools/README.md]: Sure make sense. I removed the help and added additional context.

File: isis/src/system/apps/isisdataeval/tools/README.md (hunk 2 above)
Comment (KrisBecker) [reply isis/src/system/apps/isisdataeval/tools/README.md]: Done.

**Comment by KrisBecker** [comment]
Any suggestions on where/how to install isisdata_mockup.py? This should be conveniently available to ISIS users if you decide to use this script as mechanism to validate ISISDATA installations.

**Comment by Kelvinrr** [comment]
@KrisBecker at the end of the CMakeLists.txt, you can add a line to install it to $INSTALL_PREFIX/bin/ with the .py stripped so we can run it as a command. That's how we've usually done these utility scripts we want common accesss to.

**Comment by KrisBecker** [comment]
Ok, so should I scrub all the docs to refer to the renamed extension-less version of the script? At least in the READMEs and the tests scripts?

**Comment by KrisBecker** [comment]
Commits install isisdata_mockup in $ISISROOT/bin (its not in `build/bin`). Had to finesse the permissions.

I also changed the test data mocking script to call this version.

This does create an issue for most of the documentation. It may be easier to just rename the script without the extension and change all the documentation appropriately otherwise I am concerned trying to retain the extension will lead to confusion.

**Comment by Kelvinrr** [comment]
That would be helpful, yes.

**Comment by KrisBecker** [comment]
Ok, renamed isisdata_mockup and modified docs/scripts to reflect the change.

This PR should be good to go now.

**Comment by KrisBecker** [comment]
Please let me know if there is anything else I need to do in order to get this PR merged.

Thank you.

File: isis/src/system/apps/isisdataeval/tools/README.md (hunk 1 above)
Comment (Kelvinrr) [reply isis/src/system/apps/isisdataeval/tools/README.md]: Looks like the full output of the help message is still there?

Comment (KrisBecker) [reply isis/src/system/apps/isisdataeval/tools/README.md]: This section was removed/updated in [this push](https://github.com/DOI-USGS/ISIS3/pull/5163/commits/498c9f4836ee882efe76a9dc9570926341c5eece) on or about April 3. You can also check the [source](https://github.com/KrisBecker/ISIS3/blob/feature/isisdataeval-update/isis/src/system/apps/isisdataeval/tools/README.md) to see the changes.

Comment (Kelvinrr) [reply isis/src/system/apps/isisdataeval/tools/README.md]: weird, idk why Github it doing me like this.

**Review by Kelvinrr — DISMISSED** [review]

**Comment by Kelvinrr** [comment]
Once merge conflicts are resolved I'll merge this.

**Comment by KrisBecker** [comment]
Merge conflict resolved in CHANGELOG.md and added an entry with these updates.

**Comment by Kelvinrr** [comment]
There is a random error after the merge conflicts were fixed in the test. 

```
The following tests FAILED:
	1528 - DemCube.FunctionalTestSpiceinitWebAndShapeModel (Failed)
```

Might be a fluke so I am rerunning Jenkins for now.

**Comment by Kelvinrr** [comment]
the error is still there, it did show up after the last commit. But it is a Spiceinit error trying to hit the web server. Might have been down at the time the test ran coincidentally. I'll run one more time, if it still fails on that test it will probably have to be debugged.

**Review by Kelvinrr — APPROVED** [review]
aaaaaay it passed.

**Pull request merged**